src/dev/create-extension.ts

```diff
@@ -0,0 +1,141 @@
+
+const EXTERNAL_EXTENSIONS_DIR = "externalExtensions";
+const TARGET_EXTENSION_DIR = "extension";
+const TEMPLATE_FILE_SUFFIX = ".template.mjs";
+const BASE_PATH = "base";
+const SOLIDITY_FRAMEWORKS_PATH = "solidity-frameworks";
+const DEPLOYED_CONTRACTS_FILE = "deployedContracts.ts";
+const YARN_LOCK_FILE = "yarn.lock";
+
+const prettyLog = {
+  info: (message: string, indent = 0) => console.log(chalk.cyan(`${"  ".repeat(indent)}${message}`)),
+  success: (message: string, indent = 0) => console.log(chalk.green(`${"  ".repeat(indent)}✔ ${message}`)),
+  warning: (message: string, indent = 0) => console.log(chalk.yellow(`${"  ".repeat(indent)}⚠ ${message}`)),
+  error: (message: string, indent = 0) => console.log(chalk.red(`${"  ".repeat(indent)}✖ ${message}`)),
+};
+
+const ncpPromise = promisify(ncp);
+
+const currentFileUrl = import.meta.url;
+const templateDirectory = path.resolve(decodeURI(fileURLToPath(currentFileUrl)), "../../templates");
+
+const parseArguments = (rawArgs: string[]) => {
+  const args = arg({}, { argv: rawArgs.slice(2) });
+  const projectPath = args._[0];
+  if (!projectPath) {
+    throw new Error("Project path is required");
+  }
+  return { projectPath };
+};
+
+const getChangedFilesFromFirstCommit = async (projectPath: string): Promise<string[]> => {
+  const { stdout: firstCommit } = await execa("git", ["rev-list", "--max-parents=0", "HEAD"], {
+    cwd: projectPath,
+  });
+  const { stdout } = await execa("git", ["diff", "--name-only", `${firstCommit.trim()}..HEAD`], {
+    cwd: projectPath,
+  });
+  return stdout.split("\n").filter(Boolean);
+};
+
+const getChangedFiles = async (projectPath: string): Promise<string[]> => {
+  return getChangedFilesFromFirstCommit(projectPath);
+};
+
+const createDirectories = async (filePath: string, projectName: string) => {
+  const dirPath = path.join(EXTERNAL_EXTENSIONS_DIR, projectName, TARGET_EXTENSION_DIR, path.dirname(filePath));
+  await fs.promises.mkdir(dirPath, { recursive: true });
+};
+
+const findTemplateFiles = async (dir: string, templates: Set<string>) => {
+  const files = await fs.promises.readdir(dir, { withFileTypes: true });
+  for (const file of files) {
+    const fullPath = path.join(dir, file.name);
+    if (file.isDirectory()) {
+      await findTemplateFiles(fullPath, templates);
+    } else if (file.name.endsWith(TEMPLATE_FILE_SUFFIX)) {
+      let relativePath = path.relative(templateDirectory, fullPath).replace(new RegExp(`${TEMPLATE_FILE_SUFFIX}$`), "");
+      const pathSegments = relativePath.split(path.sep);
+
+      // Normalize the relative path by stripping the initial parts
+      if (pathSegments[0] === BASE_PATH) {
+        relativePath = pathSegments.slice(1).join(path.sep);
+      } else if (pathSegments[0] === SOLIDITY_FRAMEWORKS_PATH) {
+        const framework = pathSegments[1];
+        if (Object.values(SOLIDITY_FRAMEWORKS).includes(framework as any)) {
+          relativePath = pathSegments.slice(2).join(path.sep);
```

Comment:
when I change readme it tells me
<img width="946" alt="image" src="https://github.com/scaffold-eth/create-eth/assets/25638585/f7f4bf5e-44df-4155-9327-ba6f55aa4f4d">

But it's doesn't work

Reply:
> when I change readme it tells me <img alt="image" width="946" src="https://private-user-images.githubusercontent.com/25638585/346133709-f7f4bf5e-44df-4155-9327-ba6f55aa4f4d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjAzMzAzNTksIm5iZiI6MTcyMDMzMDA1OSwicGF0aCI6Ii8yNTYzODU4NS8zNDYxMzM3MDktZjdmNGJmNWUtNDRkZi00MTU1LTkzMjctYmE2ZjU1YWE0ZjRkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzA3VDA1MjczOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFkOGY0ZWNmMjEzNmJkZWIxYTUyZDdiN2U2ZTYzNjM2OWU0YmE0NmFlZDI1OWRkZWY0OWFhMzhjOTcxYWIwOGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.WhWyVbX_A6iGZx6YwslrZr1T9BsOBtBU7cbMpG1R7wo">
> 
> But it's doesn't work

Umm didn't get it, what doesn't works 😅?

Reply:
My bad, I placed it wrong and couldn't understand why it doesn't work 😄